src/util/assert.h

```diff
@@ -1,62 +1,71 @@
 #pragma once
-
 #include <QtDebug>
+#include <cassert>
 
+/// We previously used a custom assert implementation, so this is kept for
+/// backward compatibility. The static_cast to bool makes the compiler warn if
+/// you accidently use the assignment operator instead of a comparison.
+///
+/// This also takes care of the issue that commas anywhere in condition that
+/// are not protected by parentheses are interpreted as macro argument separators
+/// (`assert` is a function-like macro).
+#define DEBUG_ASSERT(cond) assert(static_cast<bool>(cond))
 
-static constexpr const char* kDebugAssertPrefix = "DEBUG ASSERT";
-
-inline void mixxx_noop(void) {}
-
-inline void mixxx_debug_assert(const char* assertion, const char* file, int line, const char* function) {
-    qCritical("%s: \"%s\" in function %s at %s:%d", kDebugAssertPrefix, assertion, function, file, line);
-}
-
-inline bool mixxx_maybe_debug_assert_return_true(const char* assertion, const char* file, int line, const char* function) {
-#ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
-    mixxx_debug_assert(assertion, file, line, function);
+/// This also asserts that the expression evaluates to `true`, but in case the
+/// assertion was disabled (by defining`NDEBUG`), the specified fallback code
+/// is executed instead.
+/// In most cases you should probably use this rather than DEBUG_ASSERT. Only
+/// use DEBUG_ASSERT if there is no appropriate fallback.
+#ifndef NDEBUG
+#define VERIFY_OR_DEBUG_ASSERT(cond) \
+    DEBUG_ASSERT(cond);              \
+    if (false)
+#define RELEASE_ASSERT(cond) assert(static_cast<bool>(cond))
```

Comment:
@daschuer Can we adopt this static_cast for #4131 to prevent mistakes like using an assignment operator instead of equality comparison?